### PR TITLE
Editorial: move+rework examples

### LINKED ISSUE #456: Move examples to top of document
As I'm implementing this into a fake merchant site, I find that I constantly need to look up the examples. Currently they are hard to get to. 

I would like to move them to their own section at the top.

### PR DESCRIPTION
**Pull request by marcoscaceres**
* closes #456

***
[Preview](https://s3.amazonaws.com/pr-preview/w3c/browser-payment-api/move_examples.html) | [Diff](https://s3.amazonaws.com/pr-preview/w3c/browser-payment-api/a7e1bda...748163d.html)

### CODE REVIEW

**Comment by marcoscaceres** [comment]
This is a work in progress (DON'T MERGE)... It also assumes #551 will land. 

It attempts to give more detailed examples of how to actually use the API. 

It still needs to be made more clear in the examples that the payment sheet is somewhat stateless. So if you call, `.updateWith()` and you don't re-provide *all the things*, then the whole thing changes. This was quite unexpected the first time I saw it - but makes sense - but will likely catch people out. 

Basically, developers need to build a state machine to track total, possible shipping options, and how that affects the total, etc.

**Comment by zkoch** [comment]
Yeah, adding modifiers sounds great.

File: index.html [hunk 1]
```diff
@@ -182,6 +182,254 @@
         </section>
       </section>
     </section>
+    <section class="informative">
+      <h2>
+        Examples of usage
+      </h2>
+      <p>
+        In order to use the API, the developer needs to provide and keep track
+        of an number of key pieces of information. Each of these bits of
```
Comment (bentian) [inline index.html]: s/an/a

Also 'Each of' seems redundant.

File: index.html [hunk 2]
```diff
@@ -182,6 +182,254 @@
...
+      <ul>
+        <li>The <var>methodData</var>: a sequence of <a>PaymentMethodData</a>s
+        that represents the <a>payment methods</a> that the site supports
+        (e.g., "we support card-based payments, but only Visa and MasterCard
+        credit cards.").
+        </li>
+        <li>The <var>details</var>: The details of the transaction, as a
+        <a>PaymentDetailsInit</a> dictionary. This includes total cost, and
+        optionally a list of goods or services being purchased, for physical
+        goods, any shipping options. Additionally, it can optionally include
```
Comment (bentian) [inline index.html]: s/any/and

File: index.html [hunk 3]
```diff
@@ -182,6 +182,254 @@
...
+      </h2>
+      <p>
+        In order to use the API, the developer needs to provide and keep track
+        of an number of key pieces of information. Each of these bits of
+        information are passed to the <a>PaymentRequest</a> constructor as
+        arguments, and subsequently used to update the payment request being
+        displayed to the user. Namely, these bits of information are:
+      </p>
+      <ul>
+        <li>The <var>methodData</var>: a sequence of <a>PaymentMethodData</a>s
```
Comment (bentian) [inline index.html]: nit: Capitalize 'a' to conform with the other two paragraphs.

File: index.html [hunk 4]
```diff
@@ -182,6 +182,254 @@
...
+        <li>The <var>details</var>: The details of the transaction, as a
+        <a>PaymentDetailsInit</a> dictionary. This includes total cost, and
+        optionally a list of goods or services being purchased, for physical
+        goods, any shipping options. Additionally, it can optionally include
+        "modifiers" to how payments are made. For example, "if you pay with a
+        credit card of type X, it incurs a US$3.00 processing fee".
+        </li>
+        <li>The <var>options</var>: Optionally, a list of things that the site
+        needs to deliver the good or service (e.g., for physical goods, the
+        merchant will typically need an physical address to ship to. For
```
Comment (bentian) [inline index.html]: s/an/a

File: index.html [hunk 5]
```diff
@@ -182,6 +182,254 @@
...
+          ];
+        </pre>
+      </section>
+      <section>
+        <h3>
+          The <code>details</code> argument
+        </h3>
+        <p>
+          The <var>details</var> contains information about the transaction
+          that the user is being asked to complete such as the line items in an
```
Comment (bentian) [inline index.html]: A `,` after `complete` would be be clearer:
`that the user is being asked to complete, such as the line items in an order.`

File: index.html [hunk 6]
```diff
@@ -182,6 +182,254 @@
...
+        <li>The <var>options</var>: Optionally, a list of things that the site
+        needs to deliver the good or service (e.g., for physical goods, the
+        merchant will typically need an physical address to ship to. For
+        digital goods, an email will usually suffice).
+        </li>
+      </ul>
+      <p>
+        Once a <a>PaymentRequest</a> is constructed, it's presented to the end
+        user via the <a>show()</a> method. The <a>show()</a> returns a promise
+        that, once the user confirms request for payment, results in a
```
Comment (bentian) [inline index.html]: ReSpec says error for the above 2 `show()` links.

> Found linkless element with text 'show()' but no matching <dfn>

File: index.html [hunk 7]
```diff
@@ -182,6 +182,254 @@
...
+          }
+          async function validateResponse(response) {
+            try {
+              if (await checkAllValuesAreGood(response)) {
+                await response.complete("success");
+              } else {
+                await response.complete("fail");
+              }
+            } catch (err) {
+              // Something went wrong... but don't maybe finished ok?
```
Comment (bentian) [inline index.html]: Do you mean 'maybe ok to finish'?

File: index.html [hunk 8]
```diff
@@ -182,6 +182,254 @@
...
+        credit cards.").
+        </li>
+        <li>The <var>details</var>: The details of the transaction, as a
+        <a>PaymentDetailsInit</a> dictionary. This includes total cost, and
+        optionally a list of goods or services being purchased, for physical
+        goods, any shipping options. Additionally, it can optionally include
+        "modifiers" to how payments are made. For example, "if you pay with a
+        credit card of type X, it incurs a US$3.00 processing fee".
+        </li>
+        <li>The <var>options</var>: Optionally, a list of things that the site
```
Comment (bentian) [inline index.html]: Would be good to mention PaymentOptions dictionary like the other two. For example,
`Optionally, a list of things as <a>PaymentOptions</a> that the site ...`

File: index.html [hunk 9]
```diff
@@ -182,6 +182,254 @@
...
+            ],
+            total: {
+              label: "Total due",
+              amount: { currency: "USD", value: "65.00" },
+            },
+          };
+        </pre>
+        <section>
+          <h4>
+            shipping options
```
Comment (bentian) [inline index.html]: nit: Capitalize 's'hipping

**Review by bentian — APPROVED** [review]

**Comment by marcoscaceres** [comment]
@bentian, thanks so much! great finds!

File: index.html (hunk 6 above)
Comment (marcoscaceres) [reply index.html]: Will investigate separately. I might have broken something in latest ReSpec release I did today.

Comment (marcoscaceres) [reply index.html]: Actually, found the problem :) Fixed!!!
